tensorflow_addons/metrics/r_square.py

```diff
@@ -70,23 +107,39 @@ def update_state(self, y_true, y_pred, sample_weight=None):
         sample_weight = weights_broadcast_ops.broadcast_weights(sample_weight, y_true)
 
         weighted_y_true = tf.multiply(y_true, sample_weight)
-        self.sum.assign_add(tf.reduce_sum(weighted_y_true))
-        self.squared_sum.assign_add(tf.reduce_sum(tf.multiply(y_true, weighted_y_true)))
+        self.sum.assign_add(tf.reduce_sum(weighted_y_true, axis=0))
+        self.squared_sum.assign_add(
+            tf.reduce_sum(tf.multiply(y_true, weighted_y_true), axis=0)
+        )
         self.res.assign_add(
             tf.reduce_sum(
-                tf.multiply(tf.square(tf.subtract(y_true, y_pred)), sample_weight)
+                tf.multiply(tf.square(tf.subtract(y_true, y_pred)), sample_weight),
```

Comment:
Same here. would something like `((y_true - y_pred)**2) * sample_weight` work?

Reply:
It seems it does.